src/constraints/constraints_helper.jl

```diff
@@ -0,0 +1,454 @@
+export
+    DistanceConstraint,
+    AngleConstraint,
+    disable_constrained_interactions!,
+    apply_position_constraints!,
+    apply_velocity_constraints!
+
+
+abstract type Constraint{D} end
+
+"""
+    DistanceConstraint(i, j, dist)
+
+
+    DistanceConstraint(i, j, dist)
+
+Constraint between two atoms that maintains a fixed distance between the two atoms.
+"""
+struct DistanceConstraint{D} <: Constraint{D}
+    i::Int
+    j::Int
+    dist::D
+end
+
+"""
+    AngleConstraint(i, j, k, angle_jk, dist_ij, dist_ik)
+
+Constraint between three atoms that maintains a fixed angle, angle_jk, and two bond lengths. 
+Atoms j and k must be connected to a shared central atom with fixed bond distances given by
+dist_ij and dist_ik. Internally, an `AngleConstraint` is converted to 3 distance constraints. None
+of the atoms in this constraint are allowed to be constrained with another atom not
+included in this constraint. Angle can be passed as Unitful deg/rad, or as a plain
+number in radians. You cannot constrain linear molecules like CO2.
+
+# Arguments
+- `i`, `j`, `k`: The indices of the atoms in the constraint.
+- `angle_jk`: The angle between atoms j and k in radians if not Unitful.
+- `dist_ij`: The distance between atoms i and j.
+- `dist_ik`: The distance between atoms i and k.
+
+For example, a water molecule can be defined as:
+```julia
+ac = AngleConstraint(1, 2, 3, 104.5u"°", 0.9572u"Å", 0.9572u"Å")
+ac = AngleConstraint(1, 2, 3, 104.5 * π / 180, 0.9572u"Å", 0.9572u"Å")
+```
+where atom 1 is oxygen and atoms 2/3 are hydrogen.
+"""
+struct AngleConstraint{D} <: Constraint{D}
+    i::Int # Central atom
+    j::Int
+    k::Int
+    dist_ij::D
+    dist_ik::D
+    dist_jk::D
+end
+
+function AngleConstraint(i, j, k, angle_jk, dist_ij, dist_ik)
+
+    cosθ = cos(angle_jk)
+
+    if cosθ == -1.0
+        error(ArgumentError("Cannot constrain linear molecules."))
+    end
+
+    dist_jk = sqrt((dist_ij*dist_ij) + (dist_ik*dist_ik) -
+                    (2 * dist_ij * dist_ik * cosθ))
+
+    return AngleConstraint{typeof(dist_ij)}(i, j, k, dist_ij, dist_ik, dist_jk)
+end
+
+to_distance_constraints(ac::AngleConstraint) = (
+    DistanceConstraint(ac.i, ac.j, ac.dist_ij), # arm 1
+    DistanceConstraint(ac.i, ac.k, ac.dist_ik), # arm 2
+    DistanceConstraint(ac.j, ac.k, ac.dist_jk) # angle
+) 
+
+function to_cluster(ac::AngleConstraint)
+
+    cluster = @SVector [
```

Comment:
Nothing wrong with `@SVector`, but might be better to be consistent with the rest of the code and use `SVector(...)`.